File: dynabuffers-java/src/test/kotlin/dynabuffers/usecase/Schema13Test.kt
package dynabuffers.usecase

import dynabuffers.AbstractDynabuffersTest
import dynabuffers.Dynabuffers
import dynabuffers.DynabuffersEngine
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test

class Schema13Test : AbstractDynabuffersTest() {

    lateinit var engine: DynabuffersEngine;

    @BeforeEach
    fun setUp() {
        engine = Dynabuffers.parse(Schema08Test::class.java.getResourceAsStream("/schema13.dbs"))
    }

    @Test
    fun testTypeHint() {
        val onUpdate = mapOf("message" to mapOf("type" to "query", "data" to "abc".toByteArray(), ":type" to 1))
        val result = engine.deserialize(engine.serialize(onUpdate))

        assertMap(onUpdate, result)
    }

}

File: dynabuffers-python/tests/usecase/Schema13Test.py
import os
import unittest

from antlr4 import FileStream

from dynabuffers.Dynabuffers import Dynabuffers


class Schema13Test(unittest.TestCase):
    root_dir = os.path.dirname(os.path.realpath(__file__))
    error = "int too large to convert"

    def setUp(self):
        self.engine = Dynabuffers.parse(FileStream(self.root_dir + "/schema13.dbs"))

    def test_type_hint(self):
        obj = {"message": {"type": 'query', "data": bytearray(b"abc"), ":type":1}}
        result = self.engine.deserialize(self.engine.serialize(obj))
        self.assertEqual(result, obj)


if __name__ == "__main__":
    unittest.main()
